neptune/new/internal/init_model.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright (c) 2021, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import threading
+from typing import Optional
+
+from neptune.new.attributes import constants as attr_consts
+from neptune.new.exceptions import (
+    NeedExistingModelForReadOnlyMode,
+    NeptuneWongInitParametersException,
+)
+from neptune.new.internal.backends.factory import get_backend
+from neptune.new.internal.backends.project_name_lookup import project_name_lookup
+from neptune.new.internal.backgroud_job_list import BackgroundJobList
+from neptune.new.internal.operation_processors.factory import get_operation_processor
+from neptune.new.internal.utils import verify_type
+from neptune.new.internal.utils.ping_background_job import PingBackgroundJob
+from neptune.new.model import Model
+from neptune.new.types.mode import Mode
+from neptune.new.version import version as parsed_version
+
+__version__ = str(parsed_version)
+
+_logger = logging.getLogger(__name__)
+
+
+def init_model(
+    *,
+    model: Optional[str] = None,
+    name: Optional[str] = None,
+    key: Optional[str] = None,
+    project: Optional[str] = None,
+    api_token: Optional[str] = None,
+    mode: str = Mode.ASYNC.value,
+    flush_period: float = 5,
+    proxies: Optional[dict] = None,
```

Comment:
I like the logic of `_check_for_extra_kwargs` for checking non-required kwargs. Think of it :D

Reply:
I think it'd be overkill.

We've introduced additional `**kwargs` and function for checking them in `init`, because we've broken backward compability.

I don't think if we need it for newly created function.